core/src/io/anuke/mindustry/world/blocks/power/SolarGenerator.java

```diff
@@ -7,28 +7,16 @@
 import io.anuke.ucore.util.EnumSet;
 
 public class SolarGenerator extends PowerGenerator{
-    /**
-     * power generated per frame
-     */
-    protected float generation = 0.005f;
-
     public SolarGenerator(String name){
         super(name);
         flags = EnumSet.of();
+        powerProduction = 0.005f; //power generated per frame
```

Comment:
This shouldn't be needed anymore, since power is defined in the other solar panel definitions.
Not a problem with your code specifically, but still something that shouldn't be here.